### PR TITLE
Run3-hcx303 Add a few cfi's to help processing with DDD or DD4Hep

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add a few cfi's to help processing with DDD or DD4Hep

#### PR validation:

Use the standard workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/python/hcalDDConstants_cff.py [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
 from Geometry.HcalCommonData.hcalParameters_cff             import *
 from Geometry.HcalCommonData.hcalSimulationParameters_cff   import *
-from Geometry.HcalCommonData.hcalDDDSimConstants_cfi        import *
 from Geometry.HcalCommonData.hcalSimulationConstants_cfi    import *
-from Geometry.HcalCommonData.hcalDDDRecConstants_cfi        import *
 from Geometry.HcalCommonData.caloSimulationParameters_cff   import *
+from Geometry.HcalCommonData.hcalhcalDBConstants_cff        import *
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/python/hcalDDConstants_cff.py:7]: Does `hcalhcalDBConstants_cff` exist? The `hcalhcal` part seems odd.

File: Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py [hunk 2]
```diff
@@ -2,6 +2,6 @@
 
 from Geometry.HcalCommonData.hcalParameters_cff             import *
 from Geometry.HcalCommonData.hcalSimulationParameters_cff   import *
-from Geometry.HcalCommonData.hcalDDDSimConstants_cfi        import *
 from Geometry.HcalCommonData.hcalSimulationConstants_cfi    import *
 from Geometry.HcalCommonData.caloSimulationParameters_cff   import *
+from Geometry.HcalCommonData.hcalDDDSimConstants_cfi        import *
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py:7]: Is the order of the cfi's important?

File: Geometry/HcalCommonData/python/hcalDDConstants_cff.py (hunk 1 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/python/hcalDDConstants_cff.py]: Corrected

File: Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py (hunk 2 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py:7]: No it is not.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py (hunk 2 above)
Comment (cvuosalo) [reply Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py:7]: Then, unless my eyes are failing me, there is no actual change to this file, and it should be removed from the PR.

**Comment by cvuosalo** [comment]
Testing with PR #34534 might eliminate the unit test errors.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cvuosalo @silviodonato @qliphy @perrotta Do not understand the failure in the unit test. Need your advice

**Comment by perrotta** [comment]
please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
please test
(Since CMSSW_12_0_X_2021-07-22-1700 the issue with the missing file should have been solved)

**Comment by civanch** [comment]
@cvuosalo , @srimanob, it seems that this PR is ready?

File: Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py (hunk 2 above)
Comment (srimanob) [reply Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py:7]: What is the conclusion on this question? Thanks.

**Comment by srimanob** [comment]
@civanch 
I see an unanswered question in the PR. Maybe we wait a bit on that for clarification?

File: Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py (hunk 2 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py:7]: I have the ordering to be similar in all cff files. That is why it is changed here.

**Comment by bsunanda** [comment]
I don't see why it is not approved. There are serious bug fixes in a few cff's.

**Comment by srimanob** [comment]
Hi @bsunanda 

What is the bug that you are talking about? Can we not spot it on report (maybe I miss the bug report)?
Sorry, I don't see it in the PR description. Do you mean we have it for both DDD and DD4hep?

**Comment by srimanob** [comment]
+Upgrade

File: Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py (hunk 2 above)
Comment (srimanob) [reply Geometry/HcalCommonData/python/hcalDDDSimConstants_cff.py:7]: OK from my side. This is up to @cvuosalo.

Thanks.

**Comment by cvuosalo** [comment]
+1

**Comment by perrotta** [comment]
+1
(Admittently @bsunanda the title and the description of this PR are not really explaining what is done here, which basically consists in setting `fromDD4Hep = True` in all cases in which `hcalSimulationParameters` are imported. If you agree, I'd update at least the description of the PR, for better bookeping)

**Comment by bsunanda** [comment]
@cvuosalo @perrotta @srimanob The errors were in  the 2 cff's in Configuration/Geometry/python file. They were working for DDD but not for DD4Hep

**Pull request merged**